File: samples/client/3_1_0_unit_test/java/src/main/java/org/openapijsonschematools/client/schemas/validation/MinimumValidator.java
package org.openapijsonschematools.client.schemas.validation;

import org.openapijsonschematools.client.exceptions.ValidationException;
import org.checkerframework.checker.nullness.qual.Nullable;

public class MinimumValidator implements KeywordValidator {
    @Override
    public @Nullable PathToSchemasMap validate(
        ValidationData data
    ) throws ValidationException {
        var minimum = data.schema().minimum;
        if (minimum == null) {
            return null;
        }
        if (!(data.arg() instanceof Number)) {
            return null;
        }
        String msg = "Value " + data.arg() + " is invalid because it is < the minimum of " + minimum;
        if (data.arg() instanceof Integer intArg) {
            if (intArg.compareTo(minimum.intValue()) < 0) {
                throw new ValidationException(msg);
            }
            return null;
        }
        if (data.arg() instanceof Long longArg) {
            if (longArg.compareTo(minimum.longValue()) < 0) {
                throw new ValidationException(msg);
            }
            return null;
        }
        if (data.arg() instanceof Float floatArg) {
            if (floatArg.compareTo(minimum.floatValue()) < 0) {
                throw new ValidationException(msg);
            }
            return null;
        }
        if (data.arg() instanceof Double doubleArg) {
            if (doubleArg.compareTo(minimum.doubleValue()) < 0) {
                throw new ValidationException(msg);
            }
            return null;
        }
        return null;
    }
}

File: samples/client/petstore/kotlin/src/main/kotlin/org/openapijsonschematools/client/schemas/validation/MinimumValidator.kt
package org.openapijsonschematools.client.schemas.validation

import org.openapijsonschematools.client.exceptions.ValidationException

class MinimumValidator : KeywordValidator {
    @Throws(ValidationException::class)
    override fun validate(
        data: ValidationData
    ): PathToSchemasMap? {
        val minimum: Number = data.schema.minimum ?: return null
        if (data.arg !is Number) {
            return null
        }
        val msg = "Value " + data.arg + " is invalid because it is < the minimum of " + minimum
        when (data.arg) {
            is Int -> {
                if (data.arg.compareTo(minimum.toInt()) < 0) {
                    throw ValidationException(msg)
                }
            }

            is Long -> {
                if (data.arg.compareTo(minimum.toLong()) < 0) {
                    throw ValidationException(msg)
                }
            }

            is Float -> {
                if (data.arg.compareTo(minimum.toFloat()) < 0) {
                    throw ValidationException(msg)
                }
            }

            is Double -> {
                if (data.arg.compareTo(minimum.toDouble()) < 0) {
                    throw ValidationException(msg)
                }
            }
        }
        return null
    }
}
